### PR TITLE
[GCI] Add group management system tests

### PR DESCRIPTION
**Pull request by junron**
#### Describe the changes you have made in this pr -
Added system tests for group management system
Tests creating group, and editing group name, adding members and removing members!

[GCI task](https://codein.withgoogle.com/dashboard/task-instances/6398169206751232/)

### CODE REVIEW

**Comment by junron** [comment]
The error that fails the test seems to be unrelated. Also the tests worm on circle ci for some reason.
Please have a look @tachyons

**Comment by tachyons** [comment]
App veyor is windows CI, so environment can vary, we need to get this work on both platforms

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -66,6 +66,8 @@
   config.filter_rails_from_backtrace!
   # arbitrary gems may also be filtered via:
   # config.filter_gems_from_backtrace("gem name")
+
+  config.include Warden::Test::Helpers
```
Comment (tachyons) [inline spec/rails_helper.rb]: https://github.com/CircuitVerse/CircuitVerse/blob/bfe6a5884b65eca5b7de58ddd328dd39d4a13a73/spec/rails_helper.rb#L43 

Adding system here should work

Comment (junron) [reply spec/rails_helper.rb]: hmm it still doesn't work

Comment (junron) [reply spec/rails_helper.rb]: @tachyons

**Comment by junron** [comment]
@tachyons is it ok if I just do the adding group test? the adding member part seems problematic due to the morals.

**Comment by junron** [comment]
@tachyons Seems like it's more complicated than I thought, could the task be extended til Sunday?

File: spec/system/group_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe "Group management", type: :system, js: true do
+  before do
+    Capybara.raise_server_errors = false
```
Comment (Ananthavijay) [inline spec/system/group_spec.rb]: Is using this fine? @junron
Instead, just move the ``time.js`` to ``public/js`` directory. That will solve the issue I guess.

Comment (junron) [reply spec/system/group_spec.rb]: Great suggestion! I'll try it out

**Comment by junron** [comment]
@tachyons @Ananthavijay Copying the file solved the issue, but it also caused a lot of duplication. Maybe another PR could solve the underlying issue of the file not being served properly, then I'll update my PR accordingly.

**Comment by tachyons** [comment]
Only group creation is included in this PR so far, Need group managing group feature as well, like editing, deleting, adding members etc

**Comment by junron** [comment]
@tachyons @ankitkataria All done!

File: spec/system/group_spec.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    click_on "Remove"
+
+    accept_alert
+
+    expect(page).to have_text("Group member was successfully destroyed.")
+  end
+
+  it "should change the group name" do
+    visit "/groups/#{@group.id}"
+
```
Comment (alanjsph) [inline spec/system/group_spec.rb]: Is there a particular reason why you are leaving spaces between each live ?

Comment (junron) [reply spec/system/group_spec.rb]: Not really, except to separate actions from assertions. I'll remove all other unnecessary line breaks

**Comment by junron** [comment]
@alanjsph Please have a look

**Pull request merged**